### PR TITLE
feat: add test suite with GitHub Actions workflow and React version compatibility tests

### PR DESCRIPTION
**Pull request by davemecha**
## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive testing infrastructure with Jest, React Testing Library, and multi-version React compatibility.
  * Added scripts for running tests and coverage reports across React 16–19.
  * Enforced global test coverage thresholds.

* **Tests**
  * Added thorough test suites for `useAsyncEffekt` and `useAsyncMemo` hooks, covering various scenarios and edge cases.
  * Added tests verifying main module exports.
  * Provided test utilities for React hook testing compatible with multiple React versions.

* **Chores**
  * Set up automated GitHub Actions workflows for testing and linting on push and pull requests.
  * Included a test setup file to enhance Jest assertions and suppress specific deprecation warnings.
  * Added Docker-based test environment and script to run tests across React and Node.js version matrices.
  * Added configuration for review system base branches.

### CODE REVIEW

File: src/__tests__/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import semver from 'semver';
+
+let renderHook: any;
+let act: any;
+let waitFor: any;
```
[the remark this answers is not in the document]
Comment (davemecha) [reply src/__tests__/test-utils.ts:8]: wontfix. comment added in e789659eef6f08b8513d4837f83bfdea59611cd3

File: run-matrix-tests.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
```
[the remark this answers is not in the document]
Comment (davemecha) [reply run-matrix-tests.sh]: fixed in c2d7eaf3cb06aed26095086a5e756d36655c2f62

File: package.json [hunk 3]
```diff
@@ -12,7 +12,18 @@
...
+    "test:coverage": "jest --coverage",
+    "test:react16": "npm run test:install-react16 && npm test",
+    "test:react17": "npm run test:install-react17 && npm test",
+    "test:react18": "npm run test:install-react18 && npm test",
+    "test:react19": "npm run test:install-react19 && npm test",
+    "test:all-versions": "npm run test:react16 && npm run test:react17 && npm run test:react18 && npm run test:react19",
+    "test:install-react16": "npm install --force react@^16.8.0 react-dom@^16.8.0 @types/react@^16.8.0 @types/react-dom@^16.8.0 @testing-library/react@^12.1.5 @testing-library/react-hooks@^8.0.1",
+    "test:install-react17": "npm install --force react@^17.0.0 react-dom@^17.0.0 @types/react@^17.0.0 @types/react-dom@^17.0.0 @testing-library/react@^12.1.5 @testing-library/react-hooks@^8.0.1",
+    "test:install-react18": "npm install --force react@^18.0.0 react-dom@^18.0.0 @types/react@^18.0.0 @types/react-dom@^18.0.0 @testing-library/react@^14.1.0",
+    "test:install-react19": "npm install --force react@^19.0.0 react-dom@^19.0.0 @types/react@^19.0.0 @types/react-dom@^19.0.0 @testing-library/react@^14.1.0"
```
[the remark this answers is not in the document]
Comment (davemecha) [reply package.json]: fixed in 818c4ef9ca437046b4d2bdb67543fb0ec7ad5155

File: package.json [hunk 4]
```diff
@@ -36,8 +47,49 @@
...
+    "@testing-library/user-event": "^14.5.0",
+    "@types/jest": "^29.5.0",
     "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
+    "@types/semver": "^7.7.0",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
     "react": "^18.0.0",
+    "react-dom": "^18.0.0",
+    "semver": "^7.7.2",
```
[the remark this answers is not in the document]
Comment (davemecha) [reply package.json:61]: already fixed
